inc/class-rewrite.php

```diff
@@ -131,24 +131,36 @@ public function category_rewrite_rules() {
 		$categories = get_categories( array( 'hide_empty' => false ) );
 		if ( is_array( $categories ) && $categories !== array() ) {
 			foreach ( $categories as $category ) {
-				$category_nicename = $category->slug;
+				$category_nicename          = $category->slug;
+				$category_nicename_filtered = $this->filteredNicename( $category_nicename );
 				if ( $category->parent == $category->cat_ID ) {
 					// Recursive recursion.
 					$category->parent = 0;
 				}
 				elseif ( $taxonomy->rewrite['hierarchical'] != 0 && $category->parent !== 0 ) {
 					$parents = get_category_parents( $category->parent, false, '/', true );
 					if ( ! is_wp_error( $parents ) ) {
-						$category_nicename = $parents . $category_nicename;
+						$category_nicename          = $parents . $category_nicename;
+						//Convert encoded parts of the URI to Uppercase.
+						$category_nicename_filtered = $this->filteredNicename( $category_nicename );
```

Comment:
Instead of declaring this `$category_nicename_filtered` on line 135 and line 145 you can declare it on line 155, because that's the only location it is used.